tests/test_mux.py

```diff
@@ -450,92 +504,214 @@ def create_and_test_route(self, appdb, asicdb, dvs, dvs_route):
-                    ("ifname", "tun0,tun0")
-                ]
-              )
-
-        ps.set(rtprefix, fvs)
-
-        # Check if route was propagated to ASIC DB
-        rtkeys = dvs_route.check_asicdb_route_entries([rtprefix])
-
-        # Check for nexthop group and validate nexthop group member in asic db
-        self.check_nexthop_group_in_asic_db(asicdb, rtkeys[0], 2)
+        # add neighbors for initial state
+        for i,neighbor in enumerate(neighbors):
+            self.add_neighbor(dvs, neighbor, macs[i])
+        
+        try:
+            # toggle between states and add route in various combos of state
+            print("Testing add/remove/update of route")
+            starting_states = [(ACTIVE, ACTIVE), (ACTIVE, STANDBY), (STANDBY, ACTIVE), (STANDBY, STANDBY)]
+            for start in starting_states:
+                print("Adding route with %s: %s and %s: %s" % (mux_ports[0], start[0], mux_ports[1], start[1]))
+                self.set_mux_state(appdb, mux_ports[0], start[0])
+                self.set_mux_state(appdb, mux_ports[1], start[1])
+                self.add_route(dvs, route, neighbors)
+                if start[0] == "active":
+                    self.check_route_nexthop(dvs_route, asicdb, route, neighbors[0])
+                elif start[1] == "active":
+                    self.check_route_nexthop(dvs_route, asicdb, route, neighbors[1])
+                else:
+                    self.check_route_nexthop(dvs_route, asicdb, route, tunnel_nh_id, True)
+
+                print("Testing fdb update in %s, %s for %s" % (start[0], start[1], neighbors[0]))
+                # move neighbor 1
+                self.add_neighbor(dvs, neighbors[0], "00:aa:bb:cc:dd:ee")
+                if start[0] == "active":
+                    self.check_route_nexthop(dvs_route, asicdb, route, neighbors[0])
+                elif start[1] == "active":
+                    self.check_route_nexthop(dvs_route, asicdb, route, neighbors[1])
+                else:
+                    self.check_route_nexthop(dvs_route, asicdb, route, tunnel_nh_id, True)
+
+                # move neighbor 1 back
+                self.add_neighbor(dvs, neighbors[0], macs[i])
+
+                print("Testing fdb update in %s, %s for %s" % (start[0], start[1], neighbors[1]))
+                # move neighbor 2
+                self.add_neighbor(dvs, neighbors[0], "00:aa:bb:cc:dd:ee")
+                if start[0] == "active":
+                    self.check_route_nexthop(dvs_route, asicdb, route, neighbors[0])
+                elif start[1] == "active":
+                    self.check_route_nexthop(dvs_route, asicdb, route, neighbors[1])
+                else:
+                    self.check_route_nexthop(dvs_route, asicdb, route, tunnel_nh_id, True)
+
+                # move neighbor 2 back
+                self.add_neighbor(dvs, neighbors[0], macs[i])
+
+                self.del_route(dvs, route)
```

Comment:
Would prefer if this test case were parameterized so that we could use a fixture to do setup/teardown in case something goes wrong in the middle of the test. Depending on urgency, we can merge as-is and discuss this change in the future.

Reply:
yeah I agree with this. currently I think the try,finally block will teardown resources that I'm using in this test, but I can definitely rework it in that way.